lua/project/utils/history.lua

```diff
@@ -186,24 +179,21 @@ function History.setup_watch()
 end
 
 function History.read_history()
-    History.open_history('r', function(_, fd)
-        History.setup_watch()
+    local fd = History.open_history('r')
+    History.setup_watch()
 
-        if fd == nil then
-            return
-        end
+    if fd == nil then
+        return
+    end
 
-        uv.fs_fstat(fd, function(_, stat)
-            if stat == nil then
-                return
-            end
+    local stat = uv.fs_fstat(fd)
+    if not stat then
+        return
+    end
+    local data = uv.fs_read(fd, stat.size, -1)
```

Comment:
To follow up on https://github.com/DrKJeff16/project.nvim/pull/17/files#r2347430909 , if readfile was used here, there would be no fs_close operation necessary.